IpmiFeaturePkg/Library/PreBootSignalLibNull/PreBootSignalLibNull.c

```diff
@@ -0,0 +1,24 @@
+/** @file
+  NULL implementation of the SSIF SMBus library.
+
+  Copyright (c) Microsoft Corporation
+  PDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/BaseLib.h>
+#include <PiPei.h>
+
+/**
+  Opens the SMBus connection if needed.
+
+  @retval   EFI_SUCCESS           Them SMBus connection was successfully opened.
+**/
+EFI_STATUS
+EFIAPI
+SendPreBootSignaltoBmc (

```

Comment:
If we are going to leave this up to the platform to implement, I wonder if its redundant with InitializeIpmiTransportHardware. Or do you envision this being implemented in some standard ways similar to the transport libraries?

Reply:
Thinking about it some more, that function is called in both phases, so to be clearly PEI specific I think it may make sense to break this one out into it's own lib.

Reply:
Thinking about your comment, and I think you are correct. With the ipmi initialize, we do not need this library. for a given platform (BMC + SoC) the library provided will need to be checked/updated for each generation. A well designed PlatformIpmiInitialize library could cover all of these cases correclty.

I will update the PR to remove all this functionality.